st2common/st2common/metrics/base.py

```diff
@@ -24,11 +26,46 @@
 from st2common.util.date import get_datetime_utc_now
 from st2common.exceptions.plugins import PluginLoadError
 
+if not hasattr(cfg.CONF, 'metrics'):
+    from st2common.config import register_opts
+    register_opts()
+
+LOG = logging.getLogger(__name__)
 
 PLUGIN_NAMESPACE = 'st2common.metrics.driver'
 METRICS = None
 
 
+def format_metrics_key(action_db=None, liveaction_db=None, key=None):
```

Comment:
I'm not a big fan of this function definition (it's a bit confusing and hard to understand). Maybe it would make it more clear if it's split into multiple functions (get_metrics_key_for_action_db, get_metrics_key_for_liveaction_db).